universal-application-tool-0.0.1/app/views/BaseHtmlView.java

```diff
@@ -25,7 +27,40 @@
 public abstract class BaseHtmlView {
 
   public Tag renderHeader(String headerText) {
-    return h1(headerText);
+    return h1(headerText).withClasses(Styles.M_2);
+  }
+
+  public Tag renderLink(String text, String link) {
```

Comment:
I can get behind standardizing that everything that returns a j2html Tag shouldn't include the `render` prefix, but we probably also should move all of those out of `BaseView` and into utility classes.

Methods like `link`, `button`, and `checkbox` leave a lot of room for collisions/confusion, so we should try to avoid inheriting those from the base class.

`views.components.CommonComponent.link()`?

For the more complex ones we could definitely split them out like we did with `FieldWithLabel` but that currently has a similar issue. `FieldWithLabel.createInput(id)` should probably be `FieldWithLabel.input().setId(id)`

Reply:
> Methods like link, button, and checkbox leave a lot of room for collisions/confusion

Java will give you a warning if you override a method without the `@Override` annotation, which should help reduce confusion from inadvertent name collisions.

Reply:
I'm OK with leaving them in here or creating a separate CommonComponents class.

+1 - I really like your design for e.g., `FieldWithLabel`. That change sgtm.